scripts/zkai_native_attention_mlp_lifting_ablation_gate.py

```
@@ -0,0 +1,748 @@
+    if not isinstance(value, dict):
+        raise LiftingAblationError(f"{label} must be object")
+    return value
+
+
+def rows_by_path(accounting: dict[str, Any], expected_paths: set[str], label: str) -> dict[str, dict[str, Any]]:
+    rows = accounting.get("rows")
+    if not isinstance(rows, list):
+        raise LiftingAblationError(f"{label} rows must be a list")
+    result: dict[str, dict[str, Any]] = {}
+    for row in rows:
+        row = require_dict(row, f"{label} row")
```

Ruff (0.15.12)

[warning] `for` loop variable `row` overwritten by assignment target

(PLW2901)